cwl/src/main/scala/wdl4s/cwl/StringOrExpressionToCommandPart.scala

```diff
@@ -1,13 +1,15 @@
 package wdl4s.cwl
 
 import shapeless._
-import wdl4s.wdl.command.{CommandPart, CwlExpressionCommandPart, StringCommandPart}
+import wdl4s.wdl.command.StringCommandPart
+import wdl4s.wom.CommandPart
 
 object StringOrExpressionToCommandPart extends Poly1 {
-
+  val EcmaScriptRegex = """\$\(([^)]*)\)""".r
```

Comment:
this is supposed to be embedded in the `ECMAScriptExpression` Type and enforced as a Refined Type by Circe parsing.

Thus any match will show up in the at[ECMAScriptExpression] case.

Reply:
I haven't been able to witness that working, it appears all strings are matched to the `String` type even if they're ECMAScript expressions.
Is there any simple way to unit test this ?

Reply:
Yes unit tests plz

Reply:
This is a known issue.

There is something fancy one needs to use to enforce that things that don't compile.  

I created #123 specifically for this Regex, i.e. test as it involves some research.